base_comment_template/models/base_comment_template.py

```diff
@@ -124,38 +102,44 @@ class BaseCommentTemplate(models.Model):
         help="This comment template will be available on this models. "
         "You can see here only models allowed to set the coment template.",
     )
-
     domain = fields.Char(
         "Filter Domain",
         required=True,
         default="[]",
         help="This comment template will be available only for objects "
         "that satisfy the condition",
     )
-
-    priority = fields.Integer(
-        default=10, copy=False, help="the highest priority = the smallest number",
+    sequence = fields.Integer(
+        required=True, default=1, help="The smaller number = The higher priority"
     )
 
-    @api.constrains("domain", "priority", "model_ids", "position")
+    @api.constrains("domain", "sequence", "model_ids", "position")
     def _check_partners_in_company_id(self):
         templates = self.search([])
         for record in self:
-            other_template_same_models_and_priority = templates.filtered(
-                lambda t: t.priority == record.priority
+            other_template_same_models_and_sequence = templates.filtered(
+                lambda t: t.sequence == record.sequence
                 and set(record.model_ids).intersection(record.model_ids)
                 and t.domain == record.domain
                 and t.position == record.position
                 and t.id != record.id
             )
-            if other_template_same_models_and_priority:
+            if other_template_same_models_and_sequence:
                 raise ValidationError(
                     _(
-                        "There are other records with same models, priority, "
+                        "There are other records with same models, sequence, "
                         "domain and position."
                     )
                 )
 
+    def name_get(self):
+        """Redefine the name_get method to show the template name with the position.
+        """
+        res = []
+        for item in self:
+            res.append((item.id, "{} ({})".format(item.name, item.position)))
```

Comment:
I'm thinking this should include the model as well (model name, not model technical name), for being selected in partners knowing to which document they applies, but only do this through a context that is injected in the partner form, not in the other views.